### PR TITLE
Update Windows.Sys.LoggedInUsers.yaml

### PR DESCRIPTION
**Pull request by zaneGittins**
Made it more readable by using functions. Fixed performance issues by materializing queries. We also make an additional wmi call to Win32_SessionProcess which lets us filter out inactive sessions that have no processes.

### CODE REVIEW

File: content/exchange/artifacts/Windows.Sys.LoggedInUsers.yaml [hunk 1]
```diff
@@ -25,9 +25,29 @@ sources:
...
-      FROM wmi(query="SELECT * FROM Win32_LogonSession", namespace="ROOT/CIMV2") WHERE LogonId=CurrentLogonId AND Domain =~ DomainRegex AND LogonType =~ LogonTypeRegex
-
-      SELECT * FROM foreach(row=CurrentlyLoggedIn, query=Sessions)
+        LET ExtractDomain(Antecedent) = parse_string_with_regex(string=Antecedent,regex=['Domain=\\"(.*?)\\"']).g1
+        
+        LET ExtractLogonName(Antecedent) = parse_string_with_regex(string=Antecedent,regex=['Name=\\"(.*)\\"']).g1
+        
+        LET ExtractLogonID(data) = parse_string_with_regex(string=data,regex=['LogonId=\\"([0-9]+)\\"']).g1
+        
+        LET FormatTime(StartTime) = timestamp(string=parse_string_with_regex(string=StartTime,regex=['([0-9\\.]+).*']).g1+format(args=atoi(string=parse_string_with_regex(string=StartTime,regex=['(-|\\+)([0-9]+)']).g2)/60,format=parse_string_with_regex(string=StartTime,regex=['.*(\\+|-)']).g1+"%02.0f00"),format=TimeFormat)
```
Comment (scudette) [inline content/exchange/artifacts/Windows.Sys.LoggedInUsers.yaml]: This should not be necessary. The timestamp function accepts a format string to control parsing

**Comment by scudette** [comment]
Would be nicer to format the queries with proper indents. You can hit the auto format button in the GUI to get it looking fairly close

File: content/exchange/artifacts/Windows.Sys.LoggedInUsers.yaml (hunk 1 above)
Comment (zaneGittins) [reply content/exchange/artifacts/Windows.Sys.LoggedInUsers.yaml]: I think we can get pretty close using this:
```
LET TimeFormat = "20060102150405.999999-0700"
timestamp(string=regex_replace(source=Time,replace="-0",re="-"),format=TimeFormat)
```

But this format `20241026201539.494314-420` doesn't seem to be easy to parse with Go's time.Parse, specifically the -420 minutes offset.

Comment (scudette) [reply content/exchange/artifacts/Windows.Sys.LoggedInUsers.yaml]: You are right - this non standard time format is really a pain to work with. I came up with 

```sql
LET _X(X) = parse_string_with_regex(regex="(^.+)(-\\d+)$", string=X)
LET NormalizeTime(X) = format(format="%s%03g00", args=[_X(X=X).g1, int(int=_X(X=X).g2) / 60])
LET ParseTime(X) = timestamp(string=NormalizeTime(X=X), format="20060102150405.999999-0700")

SELECT ParseTime(X="20241026201539.494314-420")
FROM scope()
```

Do you think thats more readable than what you had initially?

Comment (zaneGittins) [reply content/exchange/artifacts/Windows.Sys.LoggedInUsers.yaml]: That's much better, thanks! Just pushed a commit using this method and with indents.

**Pull request merged**